pages/products-page.ts

```diff
@@ -0,0 +1,199 @@
+import { Locator, Page } from "@playwright/test";
+import { boxedStep } from "../util/boxed-step";
+import { ProductDetailsPage } from "../pages/products-details-page";
+import { Product } from "../model/product";
+import { parsePrice, parseRatingsCount } from "../util/parsers";
+
+export class ProductsPage {
+    private readonly productElements: Locator;
+    private readonly cartWrapper: Locator;
+    private readonly cartToggleButton: Locator;
+    private readonly cartContentWrapper: Locator;
+    private readonly cartProductWrappers: Locator;
+
+    constructor(private readonly page: Page) {
+        this.productElements = page.getByRole('main').getByRole('list').locator('.product-item');
+        this.cartWrapper = this.page.locator('.page-header [data-block="minicart"]');
+        this.cartToggleButton = this.cartWrapper.getByRole('link', { name: 'My Cart' });
+        this.cartContentWrapper = this.cartWrapper.getByTestId('minicart-content-wrapper');
+        this.cartProductWrappers = this.cartContentWrapper.locator('[data-role="product-item"]');
+    }
+
+    async goto() {
+        // TODO: figure out how to choose category
+        // use either overloaded method or enum
+        await this.page.goto('https://magento.softwaretestingboard.com/women/tops-women.html');
+    }
+
+    @boxedStep
+    async getProductNames(): Promise<string[]> {
+        return this.productElements.locator('.product-item-link').allInnerTexts();
+    }
+
+    @boxedStep
+    async getProducts(): Promise<Product[]> {
```

Comment:
Simple array it's not good in case that you need to refactor that data in the future, use Promise<Array< Product >>